lib/hotp/two-factor-authenticator.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+/**
+ * Handles Multi Factor Authentication for accessing parts of the
+ * application we want to verify the client has access to.
+ *
+ */
+class TwoFactorAutenticator {
+    /**
+     * We want to promisfy all methods we can because some of them
+     * take some time to generate and we don't want code going on
+     * till we have the data we need back.
+     *
+     * @param {Object} tfa
+     * @param {Object} promise
+     */
+    constructor(tfa, promise) {
+        this.tfa = promise.promisifyAll(tfa);
+    }
+
+
+    /**
+     * Generates the secret key the client will need to enter to
+     * initiate the two factor authentication.
+     *
+     * @return {Promise}
+     */
+    generateSecret() {
+        return this.tfa.generateKeyAsync(128);
+    }
+
+
+    /**
+     * Returns the image for seeing a QR code and can be then loaded into a
+     * browser or other application which can display the image.
+     * This makes it easier for the client to set up on their device as
+     * they can just scan the code.
+     *
+     * @param {string} secretKey
+     * @param {string} email
+     * @return {Promise.<string>}
+     */
+    generateQRCode(secretKey, email) {
+        email = email ? email : "";
+
+        return this.tfa.generateGoogleQRAsync("OpenToken IO", email, secretKey);
```

Comment:
This generates an image file which can be copied and pasted into the address bar of browser or I guess parsed to generate the actual image in another application. Below is an example of what is generated. The code does not reach out to any other service to generate, it is all generated in the library.

data:image/png;base64,iVBORw0KGgoA....5CYII=